src/helpers/directionalProperty.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+const positionMap = ['top', 'right', 'bottom', 'left']
+
+function generateProperty(property: string, position: string) {
+  if (!property) return position
+  const splitPropertyName = property.split('-')
+  splitPropertyName.splice(1, 0, position)
+  return splitPropertyName.join('-')
+}
+
+function generateStyles(property: string, valuesWithDefaults: Array<string|null>) {
```

Comment:
If you set `null` as a type flow will think that's an `any`, which we need to avoid. Instead of saying that, do `param: ?string`. That'll tell flow that the param is null-able!